internal/controllers/module_nmc_reconciler.go

```diff
@@ -357,7 +362,7 @@ func (mnrh *moduleNMCReconcilerHelper) enableModuleOnNode(ctx context.Context, m
 	}
 
 	opRes, err := controllerutil.CreateOrPatch(ctx, mnrh.client, nmcObj, func() error {
-		err = mnrh.nmcHelper.SetModuleConfig(nmcObj, mld, &moduleConfig)
+		err := mnrh.nmcHelper.SetModuleConfig(nmcObj, mld, &moduleConfig)
```

Comment:
`err` inside the callback called `CreateOrPatch` and outside that callback are two different variables.
If `SetModuleConfig` returns an error, then `CreateOrPatch` likely returns that error as well. But the line
```go
if err := mnrh.nmcHelper.SetModuleConfig(nmcObj, mld, &moduleConfig); err != nil {
```
does not affect the outside `err`.

Reply:
and why do we care if it affects? eventually outside err will hold the return value of CreateAndPatch, so even if it holds the error code of SeModuleConfig, it will be for a short time, and it won't affect the correctness of the code

Reply:
Oh, I see what you mean now. Indeed we could have skipped the semicolon here - I added it out of habit because I rewrote the `if` statement to hold on one line. At best, this adds one pointer on the stack - I don't think it warrants a reverting PR 🙂

Reply:
sure , no need for additional PR. Just wanted to make sure that i read the code correctly, that there is no hidden issue that i missed that required a new err variable